File: src/main/java/com/vicen/films/application/repository/StaffRepository.java
package com.vicen.films.application.repository;

import com.vicen.films.application.commands.CreateStaffMemberCommand;
import com.vicen.films.application.queries.GetStaffMemberQuery;
import com.vicen.films.application.queries.GetStaffQuery;
import com.vicen.films.domain.people.StaffMember;

import java.util.List;
import java.util.UUID;

public interface StaffRepository {

    List<StaffMember> getStaff();

    StaffMember getStaffMember(UUID staffMemberId);

    StaffMember createStaffMember(StaffMember staffMember);

}

File: src/main/kotlin/com/vicen/films/application/repository/StaffRepository.kt
package com.vicen.films.application.repository

import com.vicen.films.domain.people.StaffMember
import java.util.UUID

interface StaffRepository {

    val staff: List<StaffMember>

    fun getStaffMember(staffMemberId: UUID): StaffMember

    fun createStaffMember(staffMember: StaffMember): StaffMember

}
